src/PatternkitDrupalTwigLib.php

```diff
@@ -195,14 +247,123 @@ protected function getRawMetadata() {
    * @return string
    *   The rendered pattern HTML.
    */
-  public function getRenderedPatternMarkup(PatternkitPattern $pattern,
-    PatternkitEditorConfig $config) {
+  public function getRenderedPatternMarkup(
+    PatternkitPattern $pattern,
+    PatternkitEditorConfig $config
+  ) {
     if (empty($pattern->filename) || empty($config->fields)) {
       return '';
     }
     $template = $pattern->filename;
     $variables = $config->fields;
-    return twig_render_template($template, $variables);
+
+    return $this->renderTwigTemplate($template, $variables);
+  }
+
+  /**
+   * Returns a singleton version of the twig template engine.
+   *
+   * @return Twig_Environment
+   *   Twig environment object.
+   *
+   * @throws \Twig_Error_Loader
+   *   Twig engine instance object.
+   */
+  public function getTwigInstance() {
+    static $twig_engine;
+
+    if (!is_object($twig_engine)) {
+      // Setup twig environment.
+      // @TODO: Properly libraryize this.
+      require_once DRUPAL_ROOT . '/sites/all/libraries/Twig/Autoloader.php';
+      Twig_Autoloader::register();
+
+      $loader = new Twig_Loader_Filesystem();
+
+      $metadata = $this->getRawMetadata();
+      foreach ($metadata as $module_name => $module) {
+        if (!empty($module->filename)) {
+          $templatesDirectory = DRUPAL_ROOT . DIRECTORY_SEPARATOR . dirname(
+              $module->filename
+            );
+
+          // Namespacing is not necessary here, as each "engine" is unique to
+          // the "library".
+          $loader->addPath($templatesDirectory);
+        }
+      }
+
+      $twig_engine = new Twig_Environment(
+        $loader,
+        array(
+          'autorender'  => (bool) variable_get('pktwig_auto_render', TRUE),
+          'autoescape'  => (bool) variable_get('pktwig_auto_escape', FALSE),
+          'auto_reload' => (bool) variable_get('pktwig_auto_reload', FALSE),
+          'cache'       => variable_get('pktwig_template_cache_path', '/tmp/twig_compilation_cache'),
+          'debug'       => (bool) variable_get('pktwig_debug', FALSE),
+        )
+      );
+    }
+
+    return $twig_engine;
+  }
+
+  /**
+   * Renders a twig template on demand.
+   *
+   * @param string $template
+   *   Template filename.
+   * @param array $variables
+   *   Variables to be assigned to template.
+   *
+   * @return string
+   *   Rendered template.
+   */
+  public function renderTwigTemplate($template, array $variables = array()) {
+
+    $content = '';
```

Comment:
Unnecessary variable declaration - remove if convenient.

Reply:
It's not unnecessary in the exception short circuit.

Reply:
scratch that, it looks like I fixed that at some point.